system-helper/flatpak-system-helper.c

```diff
@@ -1962,15 +2065,39 @@ flatpak_authorize_method_handler (GDBusInterfaceSkeleton *interface,
     }
   else if (g_strcmp0 (method_name, "InstallBundle") == 0)
     {
-      const char *path;
+      const char *path, *remote, *installation;
       guint32 flags;
+      g_autoptr(GFile) path_file = NULL;
+      g_autoptr(GVariant) metadata = NULL;
+      g_autofree char *ref = NULL;
+      g_autoptr(GError) local_error = NULL;
 
       g_variant_get_child (parameters, 0, "^&ay", &path);
       g_variant_get_child (parameters, 1, "u", &flags);
+      g_variant_get_child (parameters, 2, "&s", &remote);
+      g_variant_get_child (parameters, 3, "&s", &installation);
 
       action = "org.freedesktop.Flatpak.install-bundle";
       no_interaction = (flags & FLATPAK_HELPER_INSTALL_BUNDLE_FLAGS_NO_INTERACTION) != 0;
 
+      path_file = g_file_new_for_path (path);
+      metadata = flatpak_bundle_load (path_file, NULL, &ref, NULL, NULL, NULL,
+                                      NULL, NULL, NULL, &local_error);
+      if (metadata == NULL)
+        {
+          g_dbus_method_invocation_take_error (invocation, g_steal_pointer (&local_error));
+          return FALSE;
+        }
+
+      if (!authorize_deploy_add_polkit_details (installation, system_bus,
```

Comment:
It's strange that we're adding the details here about whether the app is appropriate for the user but in `org.freedesktop.Flatpak.rules.in` the details are only checked if the action is `app-install` not `install-bundle`. So unless I'm missing something, the user could install a bundle that is "too extreme" for them.

Reply:
This code has significantly changed in the new version of the PR. Now the checks are done at deploy time, which covers installation and upgrade of conventional apps and bundles.